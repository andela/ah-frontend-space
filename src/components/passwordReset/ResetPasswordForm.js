import React from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import Input from '../common/Input';

const ResetPasswordForm = ({
  password1, password2, onChange, loading, onSubmit,
}) => (
  <form onSubmit={onSubmit}>
    <div className="col-md-6 col-sm-12">
      <div className="form-group form-group-lg">
        <Input
          className="input-field"
          type="password"
          name="password1"
          onChange={onChange}
          placeholder="Enter password here"
          value={password1}
        />
        <Input
          className="input-field"
          type="password"
          name="password2"
          onChange={onChange}
          placeholder="Confirm password here"
          value={password2}
        />
        { loading
          ? (
            <Loader
              type="Puff"
              color="#00bfff"
              height="20"
              width="20"
            />
          )
          : (
            <Input
              type="submit"
              id="submit"
              disabled={loading}
              className="btn-signin btn-reset"
              value="Reset"
            />
          )}
      </div>
    </div>
  </form>
);


ResetPasswordForm.propTypes = {
  password1: PropTypes.string.isRequired,
  password2: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

ResetPasswordForm.defaultProps = {
};

export default ResetPasswordForm;
