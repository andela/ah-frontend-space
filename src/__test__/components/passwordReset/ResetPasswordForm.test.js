import React from 'react';
import { shallow } from 'enzyme';

import ResetPasswordForm from '../../../components/passwordReset/ResetPasswordForm';

function renderResetPasswordForm(args) {
  const defaultProps = {
    password1: '',
    password2: '',
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    loading: false,
  };
  const props = { ...defaultProps, ...args };
  return shallow(<ResetPasswordForm {...props} />);
}

describe('ResetPasswordForm', () => {
  it('should match snapshot', () => {
    const wrapper = renderResetPasswordForm();
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const wrapper = renderResetPasswordForm({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });
});
