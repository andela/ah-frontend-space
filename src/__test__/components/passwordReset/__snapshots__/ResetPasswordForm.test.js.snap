// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm should match snapshot 1`] = `ShallowWrapper {}`;

exports[`ResetPasswordForm should match snapshot 2`] = `ShallowWrapper {}`;
